engine/kv_engine.cpp

```diff
@@ -932,16 +1104,20 @@ Status KVEngine::SSetImpl(Skiplist *skiplist, const StringView &user_key,
     HashEntry hash_entry;
     DataEntry data_entry;
     auto hint = hash_table_->GetHint(collection_key);
-    std::lock_guard<SpinMutex> lg(*hint.spin);
-    Status s =
-        hash_table_->SearchForWrite(hint, collection_key, SortedDataRecord,
-                                    &entry_ptr, &hash_entry, &data_entry);
+    std::unique_lock<SpinMutex> ul(*hint.spin);
+    RAIICaller local_snapshot_holder(
+        [&]() { version_controller_.HoldLocalSnapshot(); },
```

Comment:
Create a token class to perform RAII

Reply:
I think a tool class is better than tokens everywhere